clang/lib/Sema/SemaDecl.cpp

```diff
@@ -16155,7 +16155,10 @@ Decl *Sema::ActOnFinishFunctionBody(Decl *dcl, Stmt *Body,
     // This is meant to pop the context added in ActOnStartOfFunctionDef().
     ExitFunctionBodyRAII ExitRAII(*this, isLambdaCallOperator(FD));
     if (FD) {
-      FD->setBody(Body);
+      // Do not overwrite the potentially present ExtraInfo of a deleted
+      // function declaration.
+      if (!FD->isDeletedAsWritten())
+        FD->setBody(Body);
```

Comment:
When does that happen?

Reply:
`Parser::ParseFunctionDefinition()` calls `Sema::ActOnFinishFunctionBody()` *after* calling `SetFunctionBodyKind()` to set the body as deleted; with this pr, the latter also sets the deleted message which is now stored in the `DefaultedOrDeletedInfo`, but without this change, `Sema::ActOnFinishFunctionBody()` would end up overwriting it with a `nullptr` function body.

Reply:
Thanks. I think this is worth clarifying in the comment.